runtime/src/accounts_db.rs

```diff
@@ -2170,12 +2170,22 @@ impl AccountsDb {
         num_candidates
     }
 
-    pub fn shrink_all_slots(&self) {
-        for slot in self.all_slots_in_storage() {
-            if self.caching_enabled {
-                self.shrink_slot_forced(slot);
-            } else {
-                self.do_shrink_slot_forced_v1(slot);
+    pub fn shrink_all_slots(&self, startup: bool) {
+        if startup && self.caching_enabled {
```

Comment:
any particular reason we can't do the parallel shrink when caching is not enabled?

Reply:
I don't want to test that code path or be worried about it ;-)

Reply:
We are actively maintaining and improving the caching code path. I have more confidence we will get that codepath right and keep it right.